### ISSUE TITLE
Error("missing field `access_token`", line: 1, column: 40)

### ISSUE THREAD

**Issue by tobymurray** [tobymurray/mr_splashy_pants#8]
Migrated from #6 

> @tobymurray it looks like that was fixed, but now it panics on trying to access `access_token`:
> 
> ```
> thread 'main' panicked at 'Panic! reqwest::Error {
>     kind: Decode,
>     source: Error("missing field `access_token`", line: 1, column: 40),
> }', /rustc/04488afe34512aa4c33566eb16d8c912a3ae04f9/src/libstd/macros.rs:16:9
> stack backtrace:
>    0:     0x560090718f85 - backtrace::backtrace::libunwind::trace::h14d338b30b3ea0a7
>                                at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
>    1:     0x560090718f85 - backtrace::backtrace::trace_unsynchronized::h73ea91d74a3fd67f
>                                at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
>    2:     0x560090718f85 - std::sys_common::backtrace::_print_fmt::hd42948c952866e12
>                                at src/libstd/sys_common/backtrace.rs:78
>    3:     0x560090718f85 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::ha8f928866ff7571e
>                                at src/libstd/sys_common/backtrace.rs:59
>    4:     0x56009073ddac - core::fmt::write::he0c1e5f7426d2718
>                                at src/libcore/fmt/mod.rs:1076
>    5:     0x5600907139b2 - std::io::Write::write_fmt::hf3afc6cfd57d0033
>                                at src/libstd/io/mod.rs:1537
>    6:     0x56009071b620 - std::sys_common::backtrace::_print::hfc0110703f3696fd
>                                at src/libstd/sys_common/backtrace.rs:62
>    7:     0x56009071b620 - std::sys_common::backtrace::print::h3f77c6990ddfaa22
>                                at src/libstd/sys_common/backtrace.rs:49
>    8:     0x56009071b620 - std::panicking::default_hook::{{closure}}::heae49580a8d62d75
>                                at src/libstd/panicking.rs:198
>    9:     0x56009071b36c - std::panicking::default_hook::hecc34e3f729e213c
>                                at src/libstd/panicking.rs:217
>   10:     0x56009071bc63 - std::panicking::rust_panic_with_hook::he82f5d0644692441
>                                at src/libstd/panicking.rs:526
>   11:     0x56009071b85b - rust_begin_unwind
>                                at src/libstd/panicking.rs:437
>   12:     0x56009071b7cb - std::panicking::begin_panic_fmt::h905a6d44880d49ef
>                                at src/libstd/panicking.rs:391
>   13:     0x5600903d1752 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hd57cee95db9d0eaf
>   14:     0x56009040b2c0 - <async_stream::async_stream::AsyncStream<T,U> as futures_core::stream::Stream>::poll_next::h08a6deffd753621f
>   15:     0x5600903cd4e6 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h810ee83916758643
>   16:     0x5600903d26db - std::thread::local::LocalKey<T>::with::h3580e85ec25d6096
>   17:     0x5600903e1c0a - tokio::macros::scoped_tls::ScopedKey<T>::set::hbdf37f72af537fad
>   18:     0x5600903f344a - tokio::runtime::basic_scheduler::BasicScheduler<P>::block_on::hd127be604a6b9a94
>   19:     0x56009041154d - tokio::runtime::context::enter::haf2e13e89fefc440
>   20:     0x5600903f3571 - tokio::runtime::handle::Handle::enter::h3e8d78e21dd1ae45
>   21:     0x560090413ee6 - sub_watcher::main::hbec737c25c32569c
>   22:     0x5600903d25c3 - std::rt::lang_start::{{closure}}::ha7e09ce54e772948
>   23:     0x56009071c103 - std::rt::lang_start_internal::{{closure}}::h5d3ea623498f5f43
>                                at src/libstd/rt.rs:52
>   24:     0x56009071c103 - std::panicking::try::do_call::hac65e71be769a440
>                                at src/libstd/panicking.rs:348
>   25:     0x56009071c103 - std::panicking::try::hd4706e264bcf6712
>                                at src/libstd/panicking.rs:325
>   26:     0x56009071c103 - std::panic::catch_unwind::h948a0fb4a8b3ee82
>                                at src/libstd/panic.rs:394
>   27:     0x56009071c103 - std::rt::lang_start_internal::h72cc068ed2d0ac53
>                                at src/libstd/rt.rs:51
>   28:     0x560090413ff2 - main
>   29:     0x7f3c2556909b - __libc_start_main
>   30:     0x56009037518a - _start
>   31:                0x0 - <unknown>
> ```

**Comment by tobymurray** [comment]
@andrewzah - were you setting up a new client when you encountered this? I believe I've run into this in the (rather rough) initial set up, but I don't recall running into it after getting the bot's initial values configure.

**Comment by andrewzah** [comment]
This one happens roughly after 1 hour for me, consistently.

**Comment by tobymurray** [comment]
Ah - that's helpful. I imagine the access token is expiring and my renewal mechanism is either not working or unnecessarily breaking while trying to display the response. This one will take a little longer for me to fix. I'll set up a naive bot to stream continuously so maybe I can head off the next issue...

_Edit:_ Yeah, I just reproduced it (right at the hour mark). Looks like the response is `{"message": "Bad Request", "error": 400}`, which seems like I'm sending the wrong value for the refresh token when trying to refresh the access token. I'll sort out that I'm actually doing the right thing with both tokens, I'm suspicious I'm mixing them up.

**Comment by tobymurray** [comment]
I've just released 0.1.21 that, in my testing, fixes this. I had fundamentally broken the access token refresh mechanism, confusing the refresh token and the access token. This ultimately ended up overwriting the refresh token on the second access renewal (first being on application start), leading to the bad request.

Locally I've been able to stream for over an hour and have logged the access token being refreshed successfully.
